cmp/grid/columns/Column.js

```diff
@@ -243,27 +243,26 @@ export class Column {
             };
         }
 
-        // Support enhanced, absValue-aware sorting via GridSorters in GridModel.sortBy[].
-        const sortCfg = find(gridModel.sortBy, {colId: ret.colId});
-        if (sortCfg) {
-            ret.sort = sortCfg.sort;
-            ret.sortedAt = gridModel.sortBy.indexOf(sortCfg);
+        // Delegate comparator sorting to absValue-aware GridSorters in GridModel.sortBy[].
+        ret.comparator = this.comparator;
 
+        // Finally, apply explicit app requests.  The customer is always right....
+        return {...ret, ...this.agOptions};
+    }
+
+    //--------------------
+    // Implementation
+    //--------------------
+    comparator = (v1, v2, node1, node2) => {
+        const sortCfg = find(this.gridModel.sortBy, {colId: this.colId}),
             // ag-Grid sort impl. sources its primary values from the node's `groupData` property,
             // which is not what we want when sorting treeColumns.
-            if (this.isTreeColumn) {
-                ret.comparator = (v1, v2, node1, node2) => {
-                    return sortCfg.comparator(node1.data[field], node2.data[field]);
-                };
-            } else {
-                ret.comparator = (v1, v2) => sortCfg.comparator(v1, v2);
-            }
+            val1 = this.isTreeColumn ? node1.data[this.field] : v1,
+            val2 = this.isTreeColumn ? node2.data[this.field] : v2;
 
-        }
+        return sortCfg.comparator(val1, val2);
```

Comment:
How are we avoiding possible NPEs here if there is no sortCfg returned on line 257?  Is it that the comparator won't be called generally if the column isn't in the sortBy? (Assume that's the case.)

I don't want to add needlessly defensive code here, but want to ensure I understand how it's working and see if there's a way we can be a bit more explicit about it.

Reply:
Good catch. Not expecting this code to be called without a sortCfg, but probably pragmatic to fall back to a default sort (and warn?) if not found.